packages/generator/src/downloadRamlsGulpTaskHelpers.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as path from "path";
+import _ from "lodash";
+
+import { diffRaml, RamlDiff } from "@commerce-apps/raml-toolkit";
+import { generatorLogger } from "./logger";
+
+/**
+ * Extracts all the RAML file names from the specified config file.
+ *
+ * @param configPath - Target config file
+ *
+ * @returns An array of `assetId/fileName` for every api in the input file
+ */
+export function listRamlsFromConfig(configPath: string): string[] {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const config = require(configPath);
+  const ramls = [];
+  for (const apiFamily of Object.keys(config)) {
+    for (const api of config[apiFamily]) {
+      ramls.push(path.join(api.assetId, api.fatRaml.mainFile));
+    }
+  }
+
+  return ramls;
+}
+
+/**
+ * Compares the arrays to find any common or exclusive elements
+ *
+ * @param array1 - One of the arrays being compared
+ * @param array2 - The other array being compared
+ * @param common - Return all the elements that are common if true, all the
+ * elements that are not common otherwise
+ *
+ * @returns An array of RAML files that are common or not common in both the
+ * arrays
+ */
+function compareArrays(
```

Comment:
Calling this multiple times with the same args seems inefficient. Maybe do something like this?

```typescript
function split<T>(leftArr: T[], rightArr: T[]): {
  left: T[];
  right: T[];
  common: T[];
} {
  const left = new Set(leftArr);
  const right = new Set(rightArr);
  const common = new Set<T>();
  left.forEach(val => {
    if (right.has(val)) {
      left.delete(val);
      right.delete(val);
      common.add(val);
    }
  });
  return {
    left: [...left],
    right: [...right],
    common: [...common]
  };
}
```